core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -622,6 +629,10 @@ class SparkContext(config: SparkConf) extends Logging {
     _env.metricsSystem.registerSource(_dagScheduler.metricsSource)
     _env.metricsSystem.registerSource(new BlockManagerSource(_env.blockManager))
     _env.metricsSystem.registerSource(new JVMCPUSource())
+    if (executorMetricsSource != null) {
+      executorMetricsSource.register
+      env.metricsSystem.registerSource(executorMetricsSource)
```

Comment:
instead how about changing these two lines to `executorMEtricSource.register(env.metricSystem)` and pushing that second call inside.  Makes it a bit more clear for the user of a ExecutorMetricsSource